runtime/src/accounts_db.rs

```diff
@@ -965,6 +1027,20 @@ impl AccountsDB {
         }
     }
 
+    pub fn purge_slots(&self, slots: &HashSet<Slot>) {
```

Comment:
nits: How about re-implementing [the singular version](https://github.com/solana-labs/solana/pull/8799/files#diff-2099c5256db4eb5975c8834af38f6456R1022) in terms of this for less duplicate code? The singular one should only be called from `Drop for Bank`, so it's sensitive to perf.?

Reply:
Allocating the hashset is the biggest downside I think. loop setup should be pretty cheap, but you're right, performance is probably not that sensitive in this case.